index.js

```diff
@@ -148,7 +148,7 @@ function Jade () {
             }
           }
 
-          this.body = compiler(_.merge({}, defaultLocals, locals))
+          this.body = compiler(_.merge({}, defaultLocals, locals, this.state))
```

Comment:
Because of `locals` is passed to `this.render()` directly by developer, it should have higher priority than `ctx.state`, so the proper order should be: `_.merge({}, defaultLocals, this.state, locals)`.